js_modules/dagit/packages/core/src/instance/JobMenu.tsx

```diff
@@ -0,0 +1,122 @@
+import {gql, useLazyQuery} from '@apollo/client';
+import * as React from 'react';
+
+import {DISABLED_MESSAGE, usePermissions} from '../app/Permissions';
+import {canRunAllSteps, canRunFromFailure} from '../runs/RunActionButtons';
+import {RunFragments} from '../runs/RunFragments';
+import {useJobReExecution} from '../runs/useJobReExecution';
+import {ButtonWIP} from '../ui/Button';
+import {IconWIP} from '../ui/Icon';
+import {MenuWIP, MenuLink, MenuItemWIP} from '../ui/Menu';
+import {Popover} from '../ui/Popover';
+import {Tooltip} from '../ui/Tooltip';
+import {RepoAddress} from '../workspace/types';
+import {workspacePipelinePath} from '../workspace/workspacePath';
+
+import {OverviewJobFragment} from './types/OverviewJobFragment';
+import {RunReExecutionQuery} from './types/RunReExecutionQuery';
+
+interface Props {
+  job: OverviewJobFragment;
+  repoAddress: RepoAddress;
+}
+
+/**
+ * Lazily load more information about the last run for this job, then use that data to inform
+ * whether re-execution is possible.
+ */
+export const JobMenu = (props: Props) => {
+  const {job, repoAddress} = props;
+  const lastRun = job.runs[0];
```

Comment:
I am feeling inclined to remove the `"suppressImplicitAnyIndexErrors": true,` in `tsconfig`, because this isn't type-safe and one day we're gonna get bitten by it...